site/docs/v1/tech/lambdas/client-credentials-jwt-populate.adoc

```diff
@@ -0,0 +1,84 @@
+---
+layout: doc
+title: Client Credentials JWT Populate lambda
+description: An overview of the Client Credentials JWT Populate lambda
+---
+
+:sectnumlevels: 0
+
+== Client Credentials JWT Populate lambda
+
+If you would like to augment the claims provided in the JWT before it has been signed you can specify a lambda in the JWT configuration. This lambda will be invoked prior to the token being signed and issued to a user.
+
+When you create a new lambda using the FusionAuth UI we will provide you an empty function for you to implement. If you are using the API to create the lambda you will need to ensure your function has the following signature:
+
+[source,javascript]
+----
+function populate(jwt, recipientEntity, targetEntities, permissions) {
+  // Lambda code goes here
+}
+----
+
+This lambda must contain a function named `populate` that takes four parameters. The parameters that the lambda is passed are:
+
+* `jwt` - the claims object to be signed and return as the JWT payload
+* `permissions` - the permissions assigned to the Entity
+* `recipientEntity` - the Recipient Entity
+* `targetEntities` - the Target Entities
+
+The two FusionAuth objects are well documented here in the link:/docs/v1/tech/apis/entity-management/entity-types/[Permissions & Entity Types APIs] and link:/docs/v1/tech/apis/entity-management/entities/[Entities API] documentation. The JWT object is a JavaScript object containing the JWT payload. See link:/docs/v1/tech/oauth/tokens/[OpenID Connect & OAuth 2.0 Token].
+
+You may add or modify anything in the `jwt` object. However, you may not modify the header keys or values of the JWT. FusionAuth also protects certain reserved claims. The following claims are considered reserved and modifications or removal will not be reflected in the final JWT payload:
+
+- `exp`
+- `iat`
+- `sub`
+
+Prior to version 1.14.0 the following claims were considered reserved.
+
+- `applicationId`
+- `aud`
+- `authenticationType`
+- `email`
+- `email_verified`
+- `exp`
+- `iat`
+- `iss`
+- `preferred_username`
+- `roles`
+- `sub`
+
+
+== Assigning the lambda
+
+Once a lambda is created, you must assign it to an Application. See the JWT tab in the Application configuration.
+
+== Default lambda
```

Comment:
We don't ship a default lambda for this type. But when you add a new lambda of this type in the UI, you will see an example function body as you have below.